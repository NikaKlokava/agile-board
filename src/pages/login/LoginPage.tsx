import {
  signInWithPopup,
  GoogleAuthProvider,
  FacebookAuthProvider,
} from "firebase/auth";
import { useCallback, useEffect } from "react";
import { auth, googleProvider, facebookProvider } from "../../firebase";
import { LoginItem } from "../../shared/components/login_item";
import cl from "./login.module.css";

export const LoginPage = () => {
  useEffect(() => {
    window.history.replaceState("", "", "/agile-board#/authorization");
  }, []);

  const handleLoginGoogleClick = useCallback(() => {
    signInWithPopup(auth, googleProvider)
      .then((result) => {
        GoogleAuthProvider.credentialFromResult(result);
      })
      .catch((error) => {
        GoogleAuthProvider.credentialFromError(error);
      });
  }, []);

  const handleLoginFacebookClick = useCallback(async () => {
    signInWithPopup(auth, facebookProvider)
      .then((result) => {
        FacebookAuthProvider.credentialFromResult(result);
      })
      .catch((error) => {
        if (error.code === "auth/account-exists-with-different-credential")
          signInWithPopup(auth, googleProvider)
            .then((result) => {
              GoogleAuthProvider.credentialFromResult(result);
            })
            .catch((error) => {
              GoogleAuthProvider.credentialFromError(error);
            });

        FacebookAuthProvider.credentialFromError(error);
      });
  }, []);

  return (
    <div className={cl.login_wrapper}>
      <div className={cl.login_modal}>
        <h1 className={cl.title}>AGILE-BOARD</h1>
        <div className={cl.line} />
        <div className={cl.greeting_container}>
          <p className={cl.subtitle}>Hello!</p>
          <p className={cl.subtitle}>
            To continue using the board, please Log In.
          </p>
        </div>
        <div className={cl.login_container}>
          <LoginItem type="Google" onClick={handleLoginGoogleClick} />
          <LoginItem type="Facebook" onClick={handleLoginFacebookClick} />
        </div>
      </div>
    </div>
  );
};
