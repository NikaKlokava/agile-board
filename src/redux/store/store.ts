import { legacy_createStore } from "redux";
import { boardsReducer } from "../reducers/boardReducer";

const store = legacy_createStore(boardsReducer);

export default store;

// export type RootState = ReturnType<typeof store.getState>

// export interface RootState {
//     user: UserState;
//     loader: LoaderState;
//   }
// export type AppDispatch = typeof store.dispatch
