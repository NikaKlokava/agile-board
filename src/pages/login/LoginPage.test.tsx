import { Provider } from "react-redux";
import { LoginPage } from "./LoginPage";
import renderer from "react-test-renderer";
import store from "../../redux/store/store";

describe("Test the LoginPage component", () => {
  test("The LoginPage renders correctly", () => {
    const loginPageSnap = renderer
      .create(
        <Provider store={store}>
          <LoginPage />
        </Provider>
      )
      .toJSON();
    expect(loginPageSnap).toMatchSnapshot();
  });
});
