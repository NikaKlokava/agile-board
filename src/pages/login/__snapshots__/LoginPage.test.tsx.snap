// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the LoginPage component The LoginPage renders correctly 1`] = `
<div
  className="login_wrapper"
>
  <div
    className="login_modal"
  >
    <h1
      className="title"
    >
      AGILE-BOARD
    </h1>
    <div
      className="line"
    />
    <div
      className="greeting_container"
    >
      <p
        className="subtitle"
      >
        Hello!
      </p>
      <p
        className="subtitle"
      >
        To continue using the board, please Log In.
      </p>
    </div>
    <div
      className="login_container"
    >
      <div
        className="login_btn_container"
      >
        <div
          className="Google_icon"
        />
        <p
          className="login_text"
          data-testid="login_btn"
          onClick={[Function]}
        >
            Log In with Google
        </p>
      </div>
      <div
        className="login_btn_container"
      >
        <div
          className="Facebook_icon"
        />
        <p
          className="login_text"
          data-testid="login_btn"
          onClick={[Function]}
        >
            Log In with Facebook
        </p>
      </div>
    </div>
  </div>
</div>
`;
